src/foam/mlang/mlang.js

```diff
@@ -2696,6 +2701,30 @@ foam.CLASS({
 
         getArray().add(a);
       `
+    },
+    {
+      name: 'toString',
+      code: function() {
+        var projectionString = this.cls_.name + '(';
+        var exprString = [];
+        for ( var expr of this.exprs ) {
+          exprString.push(expr.toString());
+        }
+        projectionString += exprString.join(', ') + ')';
+        return projectionString;
+      },
+      javaCode: `
+        StringBuilder sb = new StringBuilder(getClassInfo().getId() + "(");
+        StringJoiner joiner = new StringJoiner(", ");
```

Comment:
You can also specify the start and end of the string int the StringJoiner which would avoid having to create the StringBuilder and having to copy the data to it.